db/views/materialized_patient_summaries_v03.sql

```diff
@@ -0,0 +1,394 @@
+        array_agg(name || '-' || dosage order by prescription_drugs.is_protocol_drug desc, prescription_drugs.name, prescription_drugs.device_created_at desc) as drug_strings
+    from blood_pressures bp
+        join prescription_drugs
+            on prescription_drugs.patient_id = bp.patient_id
+                and date(prescription_drugs.device_created_at) <= date(bp.recorded_at)
+                and (prescription_drugs.is_deleted is false or (prescription_drugs.is_deleted is true and date(prescription_drugs.device_updated_at) > date(bp.recorded_at)))
+    where bp.deleted_at is null and prescription_drugs.deleted_at is null
+    group by bp.id
+), blood_pressure_follow_up as (
+    select distinct on (bp.patient_id, date(bp.recorded_at)) bp.id bp_id, appointments.*
+    from blood_pressures bp
+    join appointments on appointments.patient_id = bp.patient_id and date(appointments.device_created_at) = date(bp.recorded_at)
+    order by bp.patient_id, date(bp.recorded_at), appointments.device_created_at desc
+), blood_sugar_follow_up as (
+    select distinct on (bs.patient_id, date(bs.recorded_at)) bs.id bs_id, appointments.*
+    from blood_sugars bs
+    join appointments on appointments.patient_id = bs.patient_id and date(appointments.device_created_at) = date(bs.recorded_at)
+    order by bs.patient_id, date(bs.recorded_at), appointments.device_created_at desc
+), ranked_blood_pressures as (
+    select
+        bp.id,
+        bp.patient_id,
+        bp.recorded_at as recorded_at,
+        bp.systolic,
+        bp.diastolic,
+        f.name as facility_name,
+        f.facility_type,
+        f.district,
+        f.state,
+        follow_up_facility.name as follow_up_facility_name,
+        a.scheduled_date as follow_up_date,
+        greatest(0, date_part('day', a.scheduled_date - date_trunc('day', a.device_created_at)))::int as follow_up_days,
+
+        bp_drugs.blood_pressure_drugs[1][1] as prescription_drug_1_name,
+        bp_drugs.blood_pressure_drugs[1][2] as prescription_drug_1_dosage,
+
+        bp_drugs.blood_pressure_drugs[2][1] as prescription_drug_2_name,
+        bp_drugs.blood_pressure_drugs[2][2] as prescription_drug_2_dosage,
+
+        bp_drugs.blood_pressure_drugs[3][1] as prescription_drug_3_name,
+        bp_drugs.blood_pressure_drugs[3][2] as prescription_drug_3_dosage,
+
+        bp_drugs.blood_pressure_drugs[4][1] as prescription_drug_4_name,
+        bp_drugs.blood_pressure_drugs[4][2] as prescription_drug_4_dosage,
+
+        bp_drugs.blood_pressure_drugs[5][1] as prescription_drug_5_name,
+        bp_drugs.blood_pressure_drugs[5][2] as prescription_drug_5_dosage,
+
+        (select string_agg(value, ', ') from unnest(bp_drugs.drug_strings[6:]) as value) as other_prescription_drugs,
+        (select string_agg(value, ', ') from unnest(bp_drugs.drug_strings) as value) as all_prescription_drugs,
+
+        rank() over (partition by bp.patient_id order by bp.recorded_at desc, bp.id) rank
```

Comment:
In very rare cases (outside of tests) it's possible that the patient has multiple blood pressures with exactly the same recorded at value. In these cases the two BPs would be assigned the same rank, resulting in multiple
 rows for the patient in generated view. 

To avoid having multiple rows with the same rank value earlier, we were doing `distinct on (patient_id, rank)` which is slow and time consuming. It is also incorrect, in that it drops the blood_pressure instead of giving it a successive rank value. 

Using the id as an additional sort parameter makes sure that rank is unique across alll BPs, and there are no duplicate patient summary rows in the result of this view.